### PR TITLE
Merge morpheus core spear phishing components.

### PR DESCRIPTION
**Pull request by drobison00**
Add Morpheus core spear phishing components.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: tests/modules/test_morpheus_modules_with_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        for _ in range(PACKET_COUNT):
+            config = {"tasks": [{"type": "load", "properties": {"loader_id": "payload", "strategy": "aggregate"}}]}
+            msg = messages.ControlMessage(config)
+            msg.payload(payload)
+            yield msg
+
+    def on_next(control_msg):
+        # pylint: disable=global-statement
+        global PACKETS_RECEIVED
+        PACKETS_RECEIVED += 1
```
Comment (mdemoret-nv) [inline tests/modules/test_morpheus_modules_with_pipeline.py:51]: Dont use global values for test assertions like this. You can define `PACKETS_RECEIVED` inside of the function and then use `nonlocal PACKETS_RECEIVED` inplace of `global PACKETS_RECEIVED`. Then each function will have a variable scoped correctly

Comment (mdemoret-nv) [reply tests/modules/test_morpheus_modules_with_pipeline.py:51]: This should be applied to all tests

File: tests/modules/test_morpheus_modules.py [hunk 2]
```diff
@@ -262,23 +262,21 @@ def test_file_loader_module():
     files = []
     file_types = ["csv", "parquet", "orc"]
     for ftype in file_types:
-        _tempfile = tempfile.NamedTemporaryFile(suffix=f".{ftype}", delete=False)
-        filename = _tempfile.name
+        with tempfile.NamedTemporaryFile(suffix=f".{ftype}", delete=False) as _tempfile:
```
Comment (mdemoret-nv) [inline tests/modules/test_morpheus_modules.py:264]: Use the pytest method for creating and working with temp files: https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

File: .gitattributes [hunk 3]
```diff
@@ -7,3 +7,4 @@ tests/mock_triton_server/payloads/** filter=lfs diff=lfs merge=lfs -text
 tests/tests_data/** filter=lfs diff=lfs merge=lfs -text
 examples/basic_usage/img/** filter=lfs diff=lfs merge=lfs -text
 docs/source/img/* filter=lfs diff=lfs merge=lfs -text
+list filter=lfs diff=lfs merge=lfs -text
```
Comment (mdemoret-nv) [inline .gitattributes:10]: Why was this added?

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
Issues found in multiple places:

1. Tests that expect an exception should use `with pytest.raises(Exception):` instead of asserting false
2. Tests should not use global variables. Instead use `nonlocal` to scope each variable to a test

File: .gitattributes (hunk 3 above)
Comment (drobison00) [reply .gitattributes:10]: @bsuryadevara ?

File: morpheus/utils/control_message_utils.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    ------
+    ValueError
+        If the payload is None.
+    """
+
+    if (control_message.payload().mutable_dataframe() is None):
+        raise ValueError("Payload cannot be None")
+
+
+def cm_default_failure_context_manager(raise_on_failure: bool = False) -> typing.Callable:
```
[the remark this answers is not in the document]
Comment (drobison00) [reply morpheus/utils/control_message_utils.py:87]: Is this necessary for this PR? If its required, it should probably go into the linter.

File: morpheus/_lib/src/modules/data_loader_module.cpp [hunk 5]
```diff
@@ -47,6 +46,11 @@ namespace morpheus {
 
 const std::string DataLoaderModule::s_config_schema = R"()";
 
+DataLoaderModule::~DataLoaderModule()
+{
+    VLOG(30) << "DataLoaderModule::~DataLoaderModule(): " << name() << std::endl;
```
[the remark this answers is not in the document]
Comment (drobison00) [reply morpheus/_lib/src/modules/data_loader_module.cpp:51]: Not really; its helpful to deal with issues related to incorrectly constructed Loaders, and its got a log level of 30.

File: morpheus/utils/execution_chain.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        for idx, function in enumerate(self.functions):
+            if not callable(function):
+                raise ValueError(f"Object {function} is not callable.")
+
+            signature = inspect.signature(function)
+
+            if (idx < len(self.functions) - 1):
+                if ((signature.return_annotation is not dict)
+                        and (typing.get_origin(signature.return_annotation) is not dict)):
+                    raise ValueError(f"Function {function.__name__} must return a dictionary. {signature}")
```
[the remark this answers is not in the document]
Comment (drobison00) [reply morpheus/utils/execution_chain.py:161]: I think you can do this as `ExecutionChain(list(function_chain))()`

[the remark this answers is not in the document]
Comment (mdemoret-nv) [reply morpheus/utils/execution_chain.py:161]: Yes you can but its a lot uglier in my opinion. A helper method here is just to make it easy to use.

File: .gitattributes (hunk 3 above)
Comment (bsuryadevara) [reply .gitattributes:10]: I'm not sure how that got into the file; possibly while I was trying to fix a gitlfs issue on the machine, that might have happened. It's not required. thanks

**Review by mdemoret-nv — APPROVED** [review]

**Comment by mdemoret-nv** [comment]
@drobison00 Make sure to link any issues that should be closed with this PR before merging
